bookstore/client/nb_client.py

```diff
@@ -126,30 +73,101 @@ def get_contents(self, path):
         return resp.json()
 
 
-def python_compat_session(session):
+class LiveNotebookServerRecord(NamedTuple):
+    """Representation of a live notebook server.
+
+    This NamedTuple is an immutable representation of the object returned by
+    ``notebook.notebookapp.list_running_servers()``.
+    """
+
+    base_url: str
+    hostname: str
+    notebook_dir: str
+    password: bool
+    pid: int
+    port: int
+    secure: bool
+    token: str
+    url: str
+
+
+def extract_kernel_id(connection_file):
+    """Helper to get a kernel id number from a filename"""
+    return os.path.basename(connection_file).lstrip('kernel-').rstrip('.json')
+
+
+class KernelInfo:
+    """Representation of Kernel information for a running kernel
+
+       Example
+       -------
+       kernel_id: str  # 'f92b7c8b-0858-4d10-903c-b0631540fb36',
+       name: str  # 'dev',
+       last_activity: str  #'2019-03-14T23:38:08.137987Z',
+       execution_state: str  #'idle',
+       connections: int  # 0 connections to the kernel
+    """
+
+    def __init__(self, kernel_id, name, last_activity, execution_state, connections):
+        # TODO: verify that *args is not needed
```

Comment:
Perhaps it should have been just a * not a *args, I was intending to make these keyword only arguments with no default values.